src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -160,23 +166,39 @@ export class RetireJsEngine implements RuleEngine {
 					// Convert the output into RuleResult objects and resolve to that.
 					res(this.processOutput(stderr, invocation.rule));
 				} else {
-					// If RetireJS exits with any other code, then it means something went wrong. We'll reject with stderr
-					// for the ease of upstream error handling.
-					rej(stderr);
+					// If RetireJS exits with any other code, then it means something went wrong. The error could be
+					// contained in either stdout or stderr, so we'll send them both to a method for processing, and
+					// reject the string we receive after prefixing it with an informative header.
+					rej(`RetireJS: ${this.processFailure(stdout, stderr)}`);
 				}
 			});
 		});
 	}
 
+	protected processFailure(stdout: string, stderr: string): string {
```

Comment:
If we are processing failure, would be helpful to have a log of the content in stdout and stderr.

Reply:
Good point. I'll add that.